### PR TITLE
fix: backend issues

### PR DESCRIPTION
**Pull request by andrewyct**
Backend issues fixed in this PR:
- #4 
- #5 
- #6 
- #7 
- #8
- #9 
- #10 
- #11 
- #12 
- #13 
- #14 
- #15 
- #16 
- #18 

Previous fixes:
- added unit tests
- updated pom.xml
- updated .gitignore

### CODE REVIEW

File: api/src/test/java/com/googlecodesamples/cloud/jss/lds/controller/FileControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+package com.googlecodesamples.cloud.jss.lds.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.googlecodesamples.cloud.jss.lds.model.BaseFile;
+import com.googlecodesamples.cloud.jss.lds.model.BaseFileTest;
+import com.googlecodesamples.cloud.jss.lds.service.FileService;
+import org.junit.Assert;
```
Comment (telpirion) [inline api/src/test/java/com/googlecodesamples/cloud/jss/lds/controller/FileControllerTest.java:9]: nit: We typically use the `com.google.common.truth` library for asserts.

See JavaDoc:
https://truth.dev/api/latest/index.html?overview-summary.html

Examples of usage:
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/speech/src/test/java/com/example/speech/AdaptationInlinePhraseSetV2IT.java#L19

File: api/src/test/java/com/googlecodesamples/cloud/jss/lds/model/BaseFileTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package com.googlecodesamples.cloud.jss.lds.model;
+
+import org.junit.Assert;
```
Comment (telpirion) [inline api/src/test/java/com/googlecodesamples/cloud/jss/lds/model/BaseFileTest.java:3]: See previous about using `com.google.common.truth` library asserts.

File: api/src/test/java/com/googlecodesamples/cloud/jss/lds/util/LdsUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.googlecodesamples.cloud.jss.lds.util;
+
+import com.googlecodesamples.cloud.jss.lds.model.BaseFileTest;
+import org.junit.Assert;
```
Comment (telpirion) [inline api/src/test/java/com/googlecodesamples/cloud/jss/lds/util/LdsUtilTest.java:4]: See previous about using `com.google.common.truth` library asserts.

**Comment by andrewyct** [comment]
New commit: 
https://github.com/GoogleCloudPlatform/app-large-data-sharing-java/pull/3/commits/e65e20c236e0cb69012489d4e580b230191326f6

Fixes:
- added com.google.truth to pom.xml
- use truth library for asserts

**Review by telpirion — APPROVED** [review]

**Comment by telpirion** [comment]
Looks good to me. Let's make sure we get the contributor's license situation all sorted out.

**Review by telpirion — APPROVED** [review]

**Pull request merged**